### DISCUSSION TITLE
GenerateResourcesAndImage is failing for Windows Server 2022

### DISCUSSION THREAD

**Discussion by Malaw80s** [actions/virtual-environments#5413] [General questions]
Hi Everyone,

I'm running into these errors below wehn running the GenerateResourcesAndImage function for Windows Server 2022.
Please see the couple of error below:

  vhd: Running tests from 'C:\image\Tests\Wix.Tests.ps1'
    vhd: Describing Wix
    vhd:   [+] Wix Toolset version from registry 689ms (486ms|203ms)
    vhd: Tests completed in 4.35s
    vhd: Tests Passed: 1, Failed: 0, Skipped: 0 NotRun: 0
==> vhd: Provisioning with powershell script: C:\virtual-environments\images\win/scripts/Installers/Install-Vsix.ps1
    vhd: Downloading package from: https://visualstudioclient.gallerycdn.vsassets.io/extensions/visualstudioclient/microsoftvisualstudio2022installerprojects/2.0.0/1641503017035/InstallerProjects.vsix to path C:\Users\packer\AppData\Local\Temp\InstallerProjects.vsix .
    vhd: Package downloaded successfully in 0.11 seconds
    vhd: Starting Install InstallerProjects.vsix...
    vhd: InstallerProjects.vsix installed successfully
    vhd: Downloading package from: https://wixtoolset.gallerycdn.vsassets.io/extensions/wixtoolset/wixtoolsetvisualstudio2022extension/1.0.0.18/1640535397051/Votive2022.vsix to path C:\Users\packer\AppData\Local\Temp\Votive2022.vsix .
    vhd: Package downloaded successfully in 0.08 seconds
    vhd: Starting Install Votive2022.vsix...
    vhd: Votive2022.vsix installed successfully
    vhd: Pester v5.3.1
    vhd:
    vhd: Starting discovery in 1 files.
    vhd: [-] Discovery in C:\image\Tests\Vsix.Tests.ps1 failed with:
    vhd: System.Net.WebException: The remote server returned an error: (503) Server Unavailable.
    vhd:    at Microsoft.PowerShell.Commands.WebRequestPSCmdlet.GetResponse(WebRequest request)
    vhd:    at Microsoft.PowerShell.Commands.WebRequestPSCmdlet.ProcessRecord()
    vhd: at Get-VsixExtenstionFromMarketplace, C:\Program Files\WindowsPowerShell\Modules\ImageHelpers\InstallHelpers.ps1: line 232
    vhd: at <ScriptBlock>, C:\image\Tests\Vsix.Tests.ps1: line 7
    vhd: at <ScriptBlock>, C:\image\Tests\Vsix.Tests.ps1: line 6
    vhd: at New-Block, C:\Program Files\WindowsPowerShell\Modules\Pester\5.3.1\Pester.psm1: line 799
    vhd: at Describe, C:\Program Files\WindowsPowerShell\Modules\Pester\5.3.1\Pester.psm1: line 9654
    vhd: at <ScriptBlock>, C:\image\Tests\Vsix.Tests.ps1: line 1
    vhd: at <ScriptBlock>, C:\Program Files\WindowsPowerShell\Modules\Pester\5.3.1\Pester.psm1: line 2985
    vhd: at Invoke-File, C:\Program Files\WindowsPowerShell\Modules\Pester\5.3.1\Pester.psm1: line 2994
    vhd: at Invoke-BlockContainer, C:\Program Files\WindowsPowerShell\Modules\Pester\5.3.1\Pester.psm1: line 2917
    vhd: at Discover-Test, C:\Program Files\WindowsPowerShell\Modules\Pester\5.3.1\Pester.psm1: line 1454
    vhd: at Invoke-Test, C:\Program Files\WindowsPowerShell\Modules\Pester\5.3.1\Pester.psm1: line 2427
    vhd: at Invoke-Pester<End>, C:\Program Files\WindowsPowerShell\Modules\Pester\5.3.1\Pester.psm1: line 5225
    vhd: at Invoke-PesterTests, C:\Program Files\WindowsPowerShell\Modules\ImageHelpers\TestsHelpers.ps1: line 67
    vhd: at <ScriptBlock>, C:\Windows\Temp\script-625de884-3c8d-b149-45b0-7032d9fe1385.ps1: line 20
    vhd: at <ScriptBlock>, <No file>: line 1
    vhd: at <ScriptBlock>, <No file>: line 1
    vhd: Discovery found 0 tests in 240.42s.
    vhd: Running tests.
    vhd: Tests completed in 240.37s
    vhd: Tests Passed: 0, Failed: 0, Skipped: 0 NotRun: 0
    vhd: Container failed: 1
    vhd:   - C:\image\Tests\Vsix.Tests.ps1
    vhd:
    vhd:
    vhd: Containers            : {[-] C:\image\Tests\Vsix.Tests.ps1}
    vhd: Result                : Failed
    vhd: FailedCount           : 0
    vhd: FailedBlocksCount     : 0
    vhd: FailedContainersCount : 1
    vhd: PassedCount           : 0
    vhd: SkippedCount          : 0
    vhd: NotRunCount           : 0
    vhd: TotalCount            : 0
    vhd: Duration              : 00:04:00.3694127
    vhd: Executed              : True
    vhd: ExecutedAt            : 4/19/2022 12:08:31 AM
    vhd: Version               : 5.3.1
    vhd: PSVersion             : 5.1.20348.643
    vhd: PSBoundParameters     : {[Configuration, PesterConfiguration]}
    vhd: Plugins               :
    vhd: PluginConfiguration   :
    vhd: PluginData            :
    vhd: Configuration         : PesterConfiguration
    vhd: DiscoveryDuration     : 00:04:00.3694127
    vhd: UserDuration          : 00:00:00
    vhd: FrameworkDuration     : 00:00:00
    vhd: Failed                : {}
    vhd: FailedBlocks          : {}
    vhd: FailedContainers      : {[-] C:\image\Tests\Vsix.Tests.ps1}
    vhd: Passed                : {}
    vhd: Skipped               : {}
    vhd: NotRun                : {}
    vhd: Tests                 : {}
    vhd: CodeCoverage          :
    vhd:
==> vhd: Test run has failed
==> vhd: At C:\Program Files\WindowsPowerShell\Modules\ImageHelpers\TestsHelpers.ps1:73 char:9
==> vhd: +         throw "Test run has failed"
==> vhd: +         ~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd:     + CategoryInfo          : OperationStopped: (Test run has failed:String) [], RuntimeException
==> vhd:     + FullyQualifiedErrorId : Test run has failed
==> vhd:
==> vhd: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?

**and when I typed R to retry I get this error:**

==> vhd: Provisioning with powershell script: C:\virtual-environments\images\win/scripts/Installers/Initialize-VM.ps1
    vhd: Disable Server Manager on Logon
    vhd:
    vhd: TaskPath                                       TaskName                          State
    vhd: --------                                       --------                          -----
    vhd: \Microsoft\Windows\Server Manager\             ServerManager                     Disabled
    vhd: Disable 'Allow your PC to be discoverable by other PCs' popup
    vhd:
    vhd: Property      : {}
    vhd: PSPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Network\NewNetw
    vhd:                 orkWindowOff
    vhd: PSParentPath  : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Network
    vhd: PSChildName   : NewNetworkWindowOff
    vhd: PSDrive       : HKLM
    vhd: PSProvider    : Microsoft.PowerShell.Core\Registry
    vhd: PSIsContainer : True
    vhd: SubKeyCount   : 0
    vhd: View          : Default
    vhd: Handle        : Microsoft.Win32.SafeHandles.SafeRegistryHandle
    vhd: ValueCount    : 0
    vhd: Name          : HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Network\NewNetworkWindowOff
    vhd:
    vhd: Disable Windows Update
    vhd: Disabled Windows Update
    vhd: Disable UAC
    vhd: User Access Control (UAC) has been disabled.
    vhd: Disable IE Welcome Screen
    vhd:
    vhd: Property      : {(default)}
    vhd: PSPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet
    vhd:                 Explorer\Main
    vhd: PSParentPath  : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet Explorer
    vhd: PSChildName   : Main
    vhd: PSDrive       : HKLM
    vhd: PSProvider    : Microsoft.PowerShell.Core\Registry
    vhd: PSIsContainer : True
    vhd: SubKeyCount   : 0
    vhd: View          : Default
    vhd: Handle        : Microsoft.Win32.SafeHandles.SafeRegistryHandle
    vhd: ValueCount    : 1
    vhd: Name          : HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet Explorer\Main
    vhd:
    vhd: Disabled IE Welcome screen
    vhd: Disable IE ESC
    vhd: IE Enhanced Security Configuration (ESC) has been disabled.
    vhd: Setting local execution policy
    vhd:
    vhd: Scope           : MachinePolicy
    vhd: ExecutionPolicy : Undefined
    vhd:
    vhd:
    vhd: Scope           : UserPolicy
    vhd: ExecutionPolicy : Undefined
    vhd:
    vhd:
    vhd: Scope           : Process
    vhd: ExecutionPolicy : Unrestricted
    vhd:
    vhd:
    vhd: Scope           : CurrentUser
    vhd: ExecutionPolicy : Undefined
    vhd:
    vhd:
    vhd: Scope           : LocalMachine
    vhd: ExecutionPolicy : Unrestricted
    vhd:
    vhd: Enable long path behavior
==> vhd: Resize-Partition : Size Not Supported
==> vhd:
==> vhd: Extended information:
==> vhd: The partition is already the requested size.
==> vhd:
==> vhd: Activity ID: {87b8b799-fb31-4e10-8e6d-24515cd4f7fd}
==> vhd: At C:\Windows\Temp\script-625de884-ae81-bbfb-b9ab-1c7cc845dbee.ps1:75 char:1
==> vhd: + Resize-Partition -DriveLetter $driveLetter -Size $size.SizeMax
==> vhd: + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd:     + CategoryInfo          : NotSpecified: (StorageWMI:ROOT/Microsoft/.../MSFT_Partition) [Resize-Partition], CimExce
==> vhd:    ption
==> vhd:     + FullyQualifiedErrorId : StorageWMI 4097,Resize-Partition
==> vhd:
==> vhd: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?

**Comment by miketimofeev** [comment]
@Malaw80s it seems that the download server was unavailable:
```
vhd: System.Net.WebException: The remote server returned an error: (503) Server Unavailable.
```
Please try to start the generation again

**Reply by Malaw80s** [reply]
Thanks @miketimofeev , when I retry I run to the error below...I have tried multiple times and it always fails at the point below:

==> vhd: Provisioning with powershell script: C:\virtual-environments\images\win/scripts/Installers/Install-Choco.ps1
    vhd: Set TLS1.2
    vhd: Install chocolatey
    vhd: WARNING: 'choco' was found at 'C:\ProgramData\Chocolatey\bin\choco.exe'.
    vhd: WARNING: An existing Chocolatey installation was detected. Installation will not continue.
    vhd: For security reasons, this script will not overwrite existing installations.
    vhd:
    vhd: Please use choco upgrade chocolatey to handle upgrades of Chocolatey itself.
    vhd: Chocolatey v1.1.0
    vhd: 2 validations performed. 1 success(es), 1 warning(s), and 0 error(s).
    vhd:
    vhd: Validation Warnings:
    vhd:  - A pending system reboot request has been detected, however, this is
    vhd:    being ignored due to the current command being used 'feature'.
    vhd:    It is recommended that you reboot at your earliest convenience.
    vhd:
    vhd: Nothing to change. Config already set.
    vhd: C:\tools
==> vhd: Provisioning with powershell script: C:\virtual-environments\images\win/scripts/Installers/Initialize-VM.ps1
    vhd: Disable Server Manager on Logon
    vhd:
    vhd: TaskPath                                       TaskName                          State
    vhd: --------                                       --------                          -----
    vhd: \Microsoft\Windows\Server Manager\             ServerManager                     Disabled
    vhd: Disable 'Allow your PC to be discoverable by other PCs' popup
    vhd:
    vhd: Property      : {}
    vhd: PSPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Network\NewNetw
    vhd:                 orkWindowOff
    vhd: PSParentPath  : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Network
    vhd: PSChildName   : NewNetworkWindowOff
    vhd: PSDrive       : HKLM
    vhd: PSProvider    : Microsoft.PowerShell.Core\Registry
    vhd: PSIsContainer : True
    vhd: SubKeyCount   : 0
    vhd: View          : Default
    vhd: Handle        : Microsoft.Win32.SafeHandles.SafeRegistryHandle
    vhd: ValueCount    : 0
    vhd: Name          : HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Network\NewNetworkWindowOff
    vhd:
    vhd: Disable Windows Update
    vhd: Disabled Windows Update
    vhd: Disable UAC
    vhd: User Access Control (UAC) has been disabled.
    vhd: Disable IE Welcome Screen
    vhd:
    vhd: Property      : {(default)}
    vhd: PSPath        : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet
    vhd:                 Explorer\Main
    vhd: PSParentPath  : Microsoft.PowerShell.Core\Registry::HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet Explorer
    vhd: PSChildName   : Main
    vhd: PSDrive       : HKLM
    vhd: PSProvider    : Microsoft.PowerShell.Core\Registry
    vhd: PSIsContainer : True
    vhd: SubKeyCount   : 0
    vhd: View          : Default
    vhd: Handle        : Microsoft.Win32.SafeHandles.SafeRegistryHandle
    vhd: ValueCount    : 1
    vhd: Name          : HKEY_LOCAL_MACHINE\Software\Policies\Microsoft\Internet Explorer\Main
    vhd:
    vhd: Disabled IE Welcome screen
    vhd: Disable IE ESC
    vhd: IE Enhanced Security Configuration (ESC) has been disabled.
    vhd: Setting local execution policy
    vhd:
    vhd: Scope           : MachinePolicy
    vhd: ExecutionPolicy : Undefined
    vhd:
    vhd:
    vhd: Scope           : UserPolicy
    vhd: ExecutionPolicy : Undefined
    vhd:
    vhd:
    vhd: Scope           : Process
    vhd: ExecutionPolicy : Unrestricted
    vhd:
    vhd:
    vhd: Scope           : CurrentUser
    vhd: ExecutionPolicy : Undefined
    vhd:
    vhd:
    vhd: Scope           : LocalMachine
    vhd: ExecutionPolicy : Unrestricted
    vhd:
    vhd: Enable long path behavior
==> vhd: Resize-Partition : Size Not Supported
==> vhd:
==> vhd: Extended information:
==> vhd: The partition is already the requested size.
==> vhd:
==> vhd: Activity ID: {c64671f8-efed-4898-90ee-561f61d08e5a}
==> vhd: At C:\Windows\Temp\script-625e3290-eac0-0e84-1dfd-29fc0a1005ef.ps1:75 char:1
==> vhd: + Resize-Partition -DriveLetter $driveLetter -Size $size.SizeMax
==> vhd: + ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
==> vhd:     + CategoryInfo          : NotSpecified: (StorageWMI:ROOT/Microsoft/.../MSFT_Partition) [Resize-Partition], CimExce
==> vhd:    ption
==> vhd:     + FullyQualifiedErrorId : StorageWMI 4097,Resize-Partition
==> vhd:
==> vhd: Script exited with non-zero exit status: 1.Allowed exit codes are: [0]
==> vhd: Step "StepProvision" failed
==> vhd: [c] Clean up and exit, [a] abort without cleanup, or [r] retry step (build may fail even if retry succeeds)?
